pkg/webhook/servicecatalog/serviceinstance/validation/handler_denyplan.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"k8s.io/apimachinery/pkg/types"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/webhookutil"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// DenyPlanChangeIfNotUpdatable checks if Plan can be changed
+func (h *AdmissionHandler) DenyPlanChangeIfNotUpdatable(ctx context.Context, req admission.Request, si *sc.ServiceInstance, traced *webhookutil.TracedLogger) error {
+	traced.Info("Starting validation - DenyPlanChangeIfNotUpdatable")
+
+	if si.Spec.ClusterServiceClassRef == nil {
+		traced.Infof("Service class does not exist")
+		return nil // user chose a service class that doesn't exist
+	}
+
+	csc := &sc.ClusterServiceClass{}
+	if err := h.client.Get(ctx, types.NamespacedName{
+		Namespace: "",
+		Name:      si.Spec.ClusterServiceClassRef.Name,
+	}, csc); err != nil {
```

Comment:
imo using inlined error + struct in method making code less readable

please consider to switching for sth like

```
	csc := &sc.ClusterServiceClass{}
	key := types.NamespacedName{
		Namespace: "",
		Name:      si.Spec.ClusterServiceClassRef.Name,
	}
	if err := h.client.Get(ctx, key , csc); err != nil {
		traced.Infof("Could not locate service class '%v', can not determine if UpdateablePlan.", si.Spec.ClusterServiceClassRef.Name)
		return err
	}
```